pkg/cloudprovider/aws/block_store.go

```diff
@@ -205,3 +209,29 @@ func (b *blockStore) DeleteSnapshot(snapshotID string) error {
 
 	return errors.WithStack(err)
 }
+
+var ebsVolumeIDRegex = regexp.MustCompile("vol-.*")
+
+func (b *blockStore) GetVolumeID(pv runtime.Unstructured) (string, error) {
+	if !collections.Exists(pv.UnstructuredContent(), "spec.awsElasticBlockStore") {
+		return "", nil
+	}
+
+	volumeID, err := collections.GetString(pv.UnstructuredContent(), "spec.awsElasticBlockStore.volumeID")
+	if err != nil {
+		return "", err
+	}
+
+	return ebsVolumeIDRegex.FindString(volumeID), nil
+}
+
+func (b *blockStore) SetVolumeID(pv runtime.Unstructured, volumeID string) (runtime.Unstructured, error) {
+	aws, err := collections.GetMap(pv.UnstructuredContent(), "spec.awsElasticBlockStore")
+	if err != nil {
+		return nil, err
+	}
+
+	aws["volumeID"] = volumeID
+
+	return pv, nil
```

Comment:
`pv` was never changed here and `aws` is not returned?

Reply:
We have to return the full object. `aws` comes from `pv` (`collections.GetMap(pv.UnstructuredContent(), "spec.awsElasticBlockStore")`). When we mutate `aws`, it's mutating `pv`.